### PR TITLE
FE magazine article search #33

### LINKED ISSUE #33: FR: Styling for FE of VT Magazine
### Description
As a business owner, I want to add a search and filtering category tag functionality to my magazine so that users can easily search, filter, and locate specific topics, articles, news, truck models, etc. This will enhance content organization to improve the user experience.

### Acceptance Criteria
**User AC:**

- [ ] There is a prominently displayed search icon that will activate a dedicated search for magazine.
- [ ] When users interact with secondary search bar on desktop/tablet devices by selecting "search magazine" or "All Topics" it shall open a dropdown box on the same page, with the names of the topic categories the user can select to filter content.
- [ ] When users interact with secondary search bar on the mobile device, this shall open the bottom drawer to full screen mode until a user selects a category or clicks the exit icon.
- [ ] Users can only select one filtering category tag option at a time.
- [ ] Once filtering options are selected on the Magazine home page, the page shall dynamically update to show the relevant articles based on tags, keeping the user on the same page.
- [ ] Filter tags shall be displayed as pill-shaped buttons, as seen in [the Figma designs.](https://www.figma.com/design/SDLKNiRf0s5Jbhqn9TJZ6H/Magazine?node-id=2325-11020&t=XGqKfrB8nmFf5i2v-1)
- [ ] The user shall see the following category tags:
- All Topics
- Volvo Trucks Magazine
- Design
- Product Feature
- Q & A
- Electromobility
- Autonomous Solutions
- Driver Training
- Safety
- Driver Productivity
- Fuel Efficiency
- Connectivity
- Sustainability
- Innovation
- Driver
- Product
- All Truck Series
- VNL
- VNR
- VNX
- VHD
- VAH
- VNR Electric

**Editor AC:**

- [ ] Content authors should have the ability to add or remove new filter tag categories.
- [ ] Content authors should have the ability to tag an article with any of the topic categories.
- [ ] Content authors should be able to determine whether selecting a new filter tag will open the category content in a new browser tab or keep the user on the current page.
- [ ] Can be used in the https://github.com/hlxsites/vg-volvotrucks-us/issues/560.

**Non-functional AC:**
- [ ] Limit the search so that magazine articles are the only search results displayed.
- [ ] The secondary magazine search bar shall be fixed or "sticky" to remain at the top of the page while the user scrolls down the articles and magazine feed page.
- [ ] One article can have up to 5 categories tags.
- [ ] Filter tags shall behave as buttons, when the user selects on a tag it shall bring them to whatever page is linked.
- [ ] Filter tags shall be compatible on white, black, and grey backgrounds.
- [ ] "All Topics" dropdown displays filtering options in alphabetical order on the search bar.

**Ticket in Jira:**
https://projects.netcentric.biz/browse/VOLVOFN-20

**Requirement Description in Wiki:**
https://projects.netcentric.biz/wiki/pages/viewpage.action?spaceKey=VOLVO&title=OS%3A+Pre-study%3A+Volvo-Trucks+Magazine+Filtering

**Related ticket:**
https://github.com/hlxsites/vg-volvotrucks-us/issues/561
https://github.com/hlxsites/vg-volvotrucks-us/issues/636

**Out of scope**
General [search page](https://www.volvotrucks.us/search-results) improvements.
Autocomplete for magazine search queries will be handled in a separate ticket after this ticket is delivered if needed.
Note
For inspo, we should check the bulldog magazine implementation for search to see if any functionality can be reused:
https://github.com/hlxsites/vg-macktrucks-com/issues/599
https://github.com/hlxsites/vg-macktrucks-com/issues/602
https://github.com/hlxsites/vg-macktrucks-com/issues/606

### Technical details
BE Search efforts are complete 

### Attachment / link to Figma
**Desktop + Tablet**
When the user first comes to the web page on desktop/tablet, they shall see the following: [Figma Link](https://www.figma.com/design/SDLKNiRf0s5Jbhqn9TJZ6H/Magazine?node-id=2325-11020&t=XGqKfrB8nmFf5i2v-1)
image

When user interacts with the secondary search by clicking on the magazine search or All Topics, they shall see the following: [Figma Link](https://www.figma.com/design/SDLKNiRf0s5Jbhqn9TJZ6H/Magazine?node-id=2325-11019&t=XGqKfrB8nmFf5i2v-1)
image

**Mobile:**

When the user first comes to the web page on a mobile device, they shall see the following: [Figma Link](
https://www.figma.com/design/SDLKNiRf0s5Jbhqn9TJZ6H/Magazine?node-id=2325-11258&t=XGqKfrB8nmFf5i2v-1)
image

When user interacts by clicking on the secondary search icon or All Topics, they shall see the following: [Figma Link](https://www.figma.com/design/SDLKNiRf0s5Jbhqn9TJZ6H/Magazine?node-id=2325-11574&t=XGqKfrB8nmFf5i2v-1)
image

### PR DESCRIPTION
**Pull request by jonatan-lledo-netcentric**
Fix #33

This block can be added above a magazine article and this will open in a new tab on the magazine's main page with the URL updated with the topic selected from the dropdown `All Topics` list and the search input also opens with the search term.

if the user goes back to the article page, the dropdown keeps the bullet selected (unless the tab is closed) and if something is searched, the input is reset

2 variants can be added: _black_ and _gray_, if both are used together, gray overwrite the black variant

URL for testing:
- https://main--volvotrucks-us--volvogroup.aem.page/drafts/jlledo/magazine/magazine-article-demo
- Before: https://33-fe-magazine-article--volvotrucks-us--volvogroup.aem.page/drafts/jlledo/magazine/magazine-article-demo

### CODE REVIEW

**Review by alexiscoelho — APPROVED** [review]

File: blocks/v2-article-search/v2-article-search.js [hunk 1]
```diff
@@ -0,0 +1,273 @@
...
+`);
+
+const buildBulletList = (allTopics) => Object.keys(allTopics).map((key) => {
+  const items = allTopics[key];
+  return items.map((item) => {
+    const param = item.toLowerCase().replace(/\s/g, '-');
+    currentURL.search = magazineParam;
+    currentURL.searchParams.set(key, param);
+    return `<li class="${blockName}__filter-item">
+      <a href="${currentURL.href.replace('#', '')}" target="_blank"
```
Comment (santi-homps) [inline blocks/v2-article-search/v2-article-search.js:99]: should this always open in a new tab? 
I see this AC: 
_Content authors should be able to determine whether selecting a new filter tag will open the category content in a new browser tab or keep the user on the current page._

Also, when going back to the original page, the topic remains selected in black

File: blocks/v2-article-search/v2-article-search.css [hunk 2]
```diff
@@ -0,0 +1,221 @@
...
+.v2-article-search__filter-dropdown--open .v2-article-search__filter-dropdown-icon {
+  transform: rotate(180deg);
+}
+
+.v2-article-search__filter-list-wrapper {
+  height: 100vh;
+  display: none;
+}
+
+ul.v2-article-search__filter-list {
```
Comment (santi-homps) [inline blocks/v2-article-search/v2-article-search.css:57]: Missing the box-shadow here
`box-shadow: 0px 4px 8px 0px #00000033;`

**Review by santi-homps — APPROVED** [review]

File: blocks/v2-article-search/v2-article-search.css (hunk 2 above)
Comment (jonatan-lledo-netcentric) [reply blocks/v2-article-search/v2-article-search.css:57]: added the shadow and a little aesthetic fix for moblie

File: blocks/v2-article-search/v2-article-search.js (hunk 1 above)
Comment (jonatan-lledo-netcentric) [reply blocks/v2-article-search/v2-article-search.js:99]: removed target blank

**Pull request merged**